ui/src/notification.ts

```diff
@@ -14,7 +14,10 @@ export enum Level {
 interface Notification {
   id: ID;
   level: Level;
+  showIcon: boolean;
   message: string;
+  actionText: string;
+  actionHandler: () => void;
```

Comment:
This action handler thing seems overly permissive. What kind of things should it be able to do? 

For example, I see that in one instance it can `openPath(path)` so should it be able to take some data and interpret it, such as `Path(string)` and interprets it as `Path(path) -> openPath(path)`.

Reply:
> What kind of things should it be able to do?

I think it could be anything. In this case we need it to fire the native method to open the OS file browser, but I could imagine it could be used for things like navigating to a certain screen in the app, etc.

<img width="512" alt="Screenshot 2020-08-11 at 09 12 56" src="https://user-images.githubusercontent.com/158411/89868477-271c0c80-dbb3-11ea-903c-745ec3f00148.png">

I'm not sure what we would gain by restricting it now, without knowing what other uses there will be in the future?

Reply:
Restricting it means we know what it does, extending the cases in the future is the [expression problem](https://en.wikipedia.org/wiki/Expression_problem).

Basically, we're giving this function too much power in the type and so it becomes harder to reason about what it does, because indeed it could do anything! Like open a file browser :scream_cat:

Again, none of my comments here are blockers but things to think about as we continue to write code and possibly make a concerted effort for the future :grin:

Reply:
> Again, none of my comments here are blockers but things to think about as we continue to write code and possibly make a concerted effort for the future 😁

I can't merge without you accepting the PR though : P

Reply:
lol fair. I wanted to provoke the conversation before you got my thumbs up :stuck_out_tongue_closed_eyes: I'm no pushover ;)

Reply:
lol